submit.py

```diff
@@ -359,15 +362,14 @@ def show_judgement(submission_url, cfg):
             if testcases_total == 0:
                 print('???', end='')
             else:
-                s = '.' * (testcases_done - 1)
+                progress = ''
+                for i in re.findall(r'<i class="([\w\- ]*)" title', status['row_html']):
+                    if 'is-empty' in i: break
+                    if 'accepted' in i: progress += color('.', _GREEN_COLOR)
+                    if 'rejected' in i: progress += color('x', _RED_COLOR)
                 if status_id == _RUNNING_STATUS:
-                    s += '?'
-                elif status_id == _ACCEPTED_STATUS:
-                    s += '.'
-                else:
-                    s += 'x'
-
-                print(f'[{s: <{testcases_total}}]  {testcases_done} / {testcases_total}', end='')
+                    progress = progress[:10*(testcases_done - 1)] + color('?', _YELLOW_COLOR)
```

Comment:
This part was a bit confusing until I realized that `len(color('.', ...))` is 10. Maybe a comment could be added that mentions that?

```suggestion
                    # N.b.: We need to do some weird math, since len(color('.', _SOME_COLOR)) = 10
                    progress = progress[:10*(testcases_done - 1)] + color('?', _YELLOW_COLOR)
```